recipes/aws-cdi-sdk/all/conanfile.py

```diff
@@ -16,6 +17,10 @@ class AwsCdiSdkConan(ConanFile):
     exports_sources = ["CMakeLists.txt", "patches/**"]
     generators = "cmake", "cmake_find_package"
     requires = "aws-libfabric/1.9.1amzncdi1.0", "aws-sdk-cpp/1.8.130"
+    default_options = {
+        "aws-libfabric:shared": True,
+        "aws-sdk-cpp:shared": True
+    }
```

Comment:
Looks like we are missing some docs... I do not know of any recipe currently that overrides defaults like this (we typically refrain from doing it in other places as well)

🟨 Can anyone confirm? i'd love to add more docs

Reply:
without it all the CI builds will get `ConanInvalidConfiguration`

Reply:
Why are static not supported?

Reply:
I tried and it doesn't compile or it crashes. I didn't invested some time in debugging it but couldn't come up with a working solution. The original package builds dynamic dependencies unconditionally: https://github.com/aws/aws-cdi-sdk/blob/a0d0cbf499e3c96ce3510810f58ae96bb6350480/Makefile#L134

Reply:
it's legal syntax and well supported. used in several places across CCI already, for instance:
https://github.com/conan-io/conan-center-index/blob/41d4fbb8c218f9d938ccb8b9d888bff52002e62e/recipes/thrust/all/test_package/conanfile.py#L8

Reply:
🤯 Awesome thanks for confirming... I never saw it being used in CCI but now I know!